### PR TITLE
Assignment 01

### PR DESCRIPTION
**Pull request by JulianFl**

### CODE REVIEW

File: src/store/useCalibrationStore.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+export const useCalibrationStore = create<CalibrationState>()(
+  persist(
+    (set) => ({
+      ...defaultCalibrationState,
+      actions: {
+        addCalibration: (start, end, steps) =>
+          set(
+            produce((state) => {
```
Comment (KiwiKilian) [inline src/store/useCalibrationStore.ts:38]: I didn't mention it in the slides, but linked it on the assignment. There is an `immer` middleware for `zustand` which makes it even easier to use, can you refactor to use it in all stores and remove the `produce` calls?

https://docs.pmnd.rs/zustand/integrations/immer-middleware

File: src/queries/useTrip.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { fetchTripHandler } from '@/functions/fetchTripHandler';
+import { LocationProps } from '@/types/Types';
+
+export function useTrip(restructureTripData: LocationProps[]) {
+  return useQuery({
+    queryKey: ['valhalla', restructureTripData],
```
Comment (KiwiKilian) [inline src/queries/useTrip.ts:8]: If you start working with entities I would recommend organizing the query keys: https://slides.kilianfinger.com/fhs-2024-ss-mmt-m-frontend-engineering/57

But for those queries which never get invalidated I don't request you to change it. Just keep in mind for maybe queries which will be added onwards.

**Review by KiwiKilian — CHANGES_REQUESTED** [review]
Looks pretty good already:

- [x] Use [`app.config.ts`](https://docs.expo.dev/guides/typescript/#appconfigjs)
- [x] Use functions from [Turf.js](https://turfjs.org/)
  - E.g. `distanceOfLatLon` to Turf.js equivalents
- [x] Make better use of `@tanstack/react-query`
  - Move mutations and queries to separate files
  - Move trip `useMutation` to `useQuery`, also others if applicable
- [x] Refactor stores:
  - [x] Rework actions, try to avoid "set" and create domain logic actions ("add", "remove", "reset"...)
  - [ ] Use [`immer` middleware](https://docs.pmnd.rs/zustand/integrations/immer-middleware) in actions
  - [x] Separate stores into trip and calibration (or similar)
  
Please make the requested changes or let me know why you would rather not like to do those changes.

File: src/queries/useTrip.ts (hunk 2 above)
Comment (JulianFl) [reply src/queries/useTrip.ts:8]: I'm not sure here. we have two queries, which query could get invalidated from your point of view?

Comment (KiwiKilian) [reply src/queries/useTrip.ts:8]: Was only a reminder for later. If you add a backend where the user profile with routes are saved you will start invalidating queries. Then it's helpful to organize the query keys.

**Review by KiwiKilian — APPROVED** [review]

**Comment by KiwiKilian** [comment]
Looks good, I will assign full points for this assignment.

Feel free to merge whenever you like.

**Pull request merged**